contracts/SmartLoan.sol

```diff
@@ -0,0 +1,459 @@
+
+  uint256 public constant LIQUIDATION_BONUS = 100;
+  uint256 private constant LIQUIDATION_CAP = 200;
+
+  uint256 public MAX_LTV = 5000;
+  uint256 public MIN_SELLOUT_LTV = 4000;
+
+  SupportedAssets supportedAssets;
+  IAssetsExchange public exchange;
+  Pool pool;
+  address private governor;
+
+  function initialize(SupportedAssets _supportedAssets, IAssetsExchange assetsExchange_, Pool pool_, address _governor) external initializer {
+    supportedAssets = _supportedAssets;
+    exchange = assetsExchange_;
+    pool = pool_;
+    __Ownable_init();
+    __PriceAware_init();
+    governor = _governor;
+    MAX_LTV = 5000;
+    MIN_SELLOUT_LTV = 4000;
+  }
+
+
+  /**
+   * Funds a loan with the value attached to the transaction
+  **/
+  function fund() external payable {
+
+    emit Funded(msg.sender, msg.value, block.timestamp);
+  }
+
+
+  function setMaxLTV(uint256 _newMaxLtv) external {
+    require(msg.sender == governor, "Only the governor account can change the maximal LTV");
+    MAX_LTV = _newMaxLtv;
+  }
+
+
+  function setMinSelloutLTV(uint256 _newMinSelloutLtv) external {
+    require(msg.sender == governor, "Only the governor account can change the minimal sellout ltv");
+    MIN_SELLOUT_LTV = _newMinSelloutLtv;
+  }
+
+
+  /**
+   * This function allows selling assets without checking if the loan will remain solvent after this operation.
+   * It is used as part of the sellout() function which sells part/all of assets in order to bring the loan back to solvency.
+   * It is possible that multiple different assets will have to be sold and for that reason we do not use the remainsSolvent modifier.
+  **/
+  function nonSolventAssetSale(bytes32 asset, uint256 _amount, uint256 _minAvaxOut) private {
+    IERC20Metadata token = getERC20TokenInstance(asset);
+    token.transfer(address(exchange), _amount);
+
+    (bool success,) = address(exchange).call{value : 0}(
+      abi.encodeWithSignature("sellAsset(bytes32,uint256,uint256)", asset, _amount, _minAvaxOut)
+    );
+    if (!success) {
+      exchange.transferBack(asset);
+    }
+  }
+
+
+  /**
+  * This function attempts to sell just enough asset to receive targetAvaxAmount.
+  * If there is not enough asset's balance to cover the whole targetAvaxAmount then the whole asset's balance
+  * is being sold.
+  * It is possible that multiple different assets will have to be sold and for that reason we do not use the remainsSolvent modifier.
+  **/
+  function nonSolventPartialOrFullAssetSale(bytes32 asset, uint256 targetAvaxAmount) private {
+    IERC20Metadata token = getERC20TokenInstance(asset);
+    uint256 balance = token.balanceOf(address(this));
+    if (balance > 0) {
+      uint256 minSaleAmount = exchange.getMinimumERC20TokenAmountForExactAVAX(targetAvaxAmount, supportedAssets.getAssetAddress(asset));
+      if (balance < minSaleAmount) {
+        uint256 saleAvaxValue = exchange.getEstimatedAVAXFromERC20Token(balance, supportedAssets.getAssetAddress(asset));
```

Comment:
If we're selling all the avax does it improve anything getting an estimate first and then using the estimate as minExpectedAmount. It seems unnecessary. 
What will be the advantage of doing that instead of trying to sell with the 0 as minAmountOut?